src/main/java/reactor/rabbitmq/OutboundMessage.java

```diff
@@ -23,7 +23,7 @@
 /**
  * Outbound message meant to be sent by a {@link Sender}.
  */
-public class OutboundMessage {
+public class OutboundMessage<T> {
```

Comment:
rather than force a `null` correlationMetadata on every user, I'd extend `OutboundMessage` and put the generic and the field on the subclass (`OutboundCorrelatedMessage`?)